documentation/developer-guide/modules/tooling-guide/pages/maven-plugin.adoc

```diff
@@ -0,0 +1,424 @@
+:page-partial:
+
+[[maven-plugin]]
+= Maven Plugin
+
+The `sds-aspect-model-maven-plugin` wraps the functionality provided by the SDS-SDK. Each piece of supported functionality is implemented as a goal. As such, developers can easily work with Aspect Models in the build of their maven projects to, for example, generate code from an Aspect Model.
+
+To include the plugin, use the following dependency:
+
+[source,maven,subs=attributes+]
+----
+<dependency>
+    <groupId>io.openmanufacturing</groupId>
+    <artifactId>sds-aspect-model-maven-plugin</artifactId>
+    <version>{sds-sdk-version}</version>
+</dependency>
+----
+
+NOTE: the current implementation of the `sds-aspect-model-maven-plugin` uses the `FileSystemStrategy` to resolve Aspect Models. See xref:java-aspect-tooling.adoc#understanding-model-resolution for more information.
+
+== Validating an Aspect Model
+
+The `validate` goal validates whether a given Aspect Model adheres to the BAMM. The default life cycle phase for the goal is `validate`.
+
+Usage:
+
+[source,maven,subs=attributes+]
+----
+<build>
+  <plugins>
+     <plugin>
+        <artifactId>sds-aspect-model-maven-plugin</artifactId>
+        <executions>
+           <execution>
+              <id>validate-aspect-model</id>
+              <goals>
+                 <goal>validate</goal>
+              </goals>
+           </execution>
+        </executions>
+        <configuration>
+           <aspectModelFilePath>{path-to-aspect-model/AspectModelFile.ttl}</aspectModelFilePath>
```

Comment:
I could imagine that one would have multiple Aspect Models in a project. I'd expect some capability to refer to multiple models using globs/wildcards, similar to how the [Maven Surefire Plugin](https://maven.apache.org/surefire/maven-surefire-plugin/examples/inclusion-exclusion.html) or the [Maven Assembly Plugin](https://maven.apache.org/plugins/maven-assembly-plugin/examples/single/including-and-excluding-artifacts.html) do it. It seems there is no unified mechanism in Maven for this, but I think it would be good idea to do it like one of the popular plugins like Surefire.

Reply:
I do not think that wildcards will be a big benefit since the file names of the Aspect Model files in most cases will not have a anything in common.
Instead we can extend the configuration to provide the root folder where the Aspect Model files reside and a list of specific Aspect Models to be included. The list of Aspect Models can be provided using the URN of the Models.